File: Array/LC_242_Valid_Anagram.cpp
// Written by: Jian-Yuan Lin
// Date: 2022.11.17

class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()){
            return false;
        }
        vector<int> s_count(26, 0), t_count(26, 0);
        for (int i = 0; i < s.size(); i++){
            s_count[s[i] - 'a']++;
            t_count[t[i] - 'a']++;
        }
        for (int i = 0; i < 26; i++){
            if (s_count[i] != t_count[i]){
                return false;
            }
        }
        return true;
    }
};

File: Array/LC_242_Valid_Anagram.py
# Written by: Jian-Yuan Lin
# Date: 2022.11.17

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        freq_s, freq_t = [0] * 26, [0] * 26
        
        if len(s) != len(t):
            return False
        n = len(s)
        
        for i in range(n):
            freq_s[ord(s[i]) - ord("a")] += 1
            freq_t[ord(t[i]) - ord("a")] += 1
        
        for i in range(26):
            if freq_s[i] != freq_t[i]:
                return False
        return True